src/ca/csr.rs

```diff
@@ -360,23 +367,33 @@ impl CsrAttributes for RpkiCaCsrAttributes {
                                         content, &mut sia
                                     )
                                 } else {
-                                    xerr!(Err(decode::Malformed))
+                                    Err(content.content_err(
+                                        InvalidExtension::new(id)
+                                    ))
                                 }
-                            })?;
+                            }).map_err(DecodeError::convert)?;
 
 
                             Ok(())
                         })? {};
                         Ok(())
                     })})
                 } else {
-                    xerr!(Err(decode::Malformed).map_err(Into::into))
+                    Err(cons.content_err(InvalidAttribute::new(id)))
                 }
             })?;
 
-            let basic_ca = basic_ca.ok_or(decode::Malformed)?;
-            let key_usage = key_usage.ok_or(decode::Malformed)?;
-            let sia = sia.ok_or(decode::Malformed)?;
+            let basic_ca = basic_ca.ok_or_else(|| {
+                cons.content_err("missing Basic Constraints extension")
```

Comment:
Below i see a dedicated error type for InvalidAttribute, why not also one for MissingExtension?

Reply:
The purpose of these dedicated error types is to avoid having to allocate the string with the error message – `InvalidAttribute` contains the OID of the attribute. They are never exposed publicly.

So, for anything where the error message is a static str, using a regular `ContentErr` is fine. Better even, since it avoid having to box the error value and just uses the str.